pysparkle/pysparkle/data_quality/data_quality_utils.py

```diff
@@ -0,0 +1,139 @@
+"""
+Data_quality utility functions to set up validations
+"""
+from typing import List, Dict
+from pyspark.sql import DataFrame
+
+from great_expectations.core.batch import RuntimeBatchRequest
+from great_expectations.core.expectation_configuration import (
+    ExpectationConfiguration)
+from great_expectations.core.expectation_validation_result import ExpectationSuiteValidationResult
+from great_expectations.core.expectation_suite import ExpectationSuite
+from great_expectations.data_context import BaseDataContext
+from great_expectations.data_context.types.base import DataContextConfig, \
+    FilesystemStoreBackendDefaults
+
+
+def create_datasource_config(datasource_name: str) -> BaseDataContext:
+    """
+    Given a string containing the datasource name, this function generates a data context instance
+
+    Args:
+        datasource_name (str): Name of the datasource to be validated
+
+    Returns:
+        BaseDataContext: populated data context instance to which expectations can be added
+    """
+
+    datasource_config = {
+        "name": f"{datasource_name}",
+        "class_name": "Datasource",
+        "module_name": "great_expectations.datasource",
+        "execution_engine": {
+            "module_name": "great_expectations.execution_engine",
+            "class_name": "SparkDFExecutionEngine",
+        },
+        "data_connectors": {
+            f"{datasource_name}_data_connector": {
+                "module_name": "great_expectations.datasource.data_connector",
+                "class_name": "RuntimeDataConnector",
+                "batch_identifiers": [
+                    "batch_name",
+                ]
+            },
+        },
+    }
+
+    root_directory = f"/dbfs/great_expectations/{datasource_name}/"
+
+    data_context_config = DataContextConfig(
+        store_backend_defaults=FilesystemStoreBackendDefaults(
+            root_directory=root_directory
+        ),
+    )
+
+    context = BaseDataContext(project_config=data_context_config)
+
+    context.add_datasource(**datasource_config)
+
+    return context
+
+
+def add_expectations_for_columns(
+    expectation_suite: ExpectationSuite,
+    validation_conf: Dict,
+) -> ExpectationSuite:
+    """Add expectations for columns as defined in the config file.
+    
+        Args:
+        expectation_suite (ExpectationSuite): Existing expectation suite to be added to
+        validation_conf (dict): dict containing details of validators to be added to columns
+
+    Returns:
+        ExpectationSuite: Expectation suite with new expectations saved
+    """
```

Comment:
```suggestion
    """Add expectations for columns as defined in the config file.
    
    Args:
        expectation_suite: Existing expectation suite to be added to
        validation_conf: dict containing details of validators to be added to columns

    Returns:
        Expectation suite with new expectations saved
    """
```
Same as in the other place: I'd rather stick to one convention: if we are already using typehints in the function definition, there is no point in repeating them in the docstrings. Docs generation/rendering tools can use the typehints.

Reply:
Fine by me